### PR TITLE
Update scikit-learn to 0.17.0

### PR DESCRIPTION
**Pull request by desilinguist**
The only change that was required was to patch up learner objects from older versions of SKLL (and scikit-learn) that we might load using the `from_file()` class method.

### CODE REVIEW

**Comment by desilinguist** [comment]
I also ran the unit tests with `gridmap` on python 2.7 and 3.3. Everything passes.

**Comment by dan-blanchard** [comment]
:+1: Looks good to merge to me.

**Comment by desilinguist** [comment]
I looked through the [release notes](http://scikit-learn.org/stable/whats_new.html) for scikit-learn 0.17.0 and made any appropriate changes. But please take a look to see if I have missed anything.

**Comment by dan-blanchard** [comment]
We should verify that we don't pass 1D arrays anywhere:

> Passing 1D data arrays as input to estimators is now deprecated as it caused confusion in how the array elements should be interpreted as features or as samples. All data arrays are now expected to be explicitly shaped (n_samples, n_features). By Vighnesh Birodkar.

We should rely on this new attribute instead of doing the subclass check we do now:

> Classifier and Regressor models are now tagged as such using the `_estimator_type` attribute.

I'd have to take a closer look if this is a problem or not for us:

> Cross-validation iterators allways provide indices into training and test set, not boolean masks.

Those are the only things that jumped out at me.

**Comment by desilinguist** [comment]
Thanks @dan-blanchard!

1. I don't see any 1D arrays being passed to any `.fit()` or `.transform()` methods for estimators.

2. I replaced all the `issubclass(..., RegressorMixin)` to check for `_estimator_type` instead.

3. We have three cross-validation iterators: `LeaveOneLabelOut`, `StratifiedKFold`, and `KFold. All of them are used in `learner.py` and all of them are being used as iterators over indices.

**Comment by AlJohri** [comment]
Can you push a minor version upgrade with this change if 1.2 is far off? Looking to try skll in my pipeline. Installing HEAD from git for now. Thanks!

`pip install "git+git://github.com/EducationalTestingService/skll.git#egg=skll"`

**Comment by desilinguist** [comment]
@AlJohri actually we are planning to release 1.2 early next week :)

**Comment by AlJohri** [comment]
:+1:
